### DISCUSSION TITLE
Automation - deletion of all records in the Budibase internal table

### DISCUSSION THREAD

**Discussion by mallipatil** [Budibase/budibase#14830] [Help] [answered]
**Context:** 
Cron job to load data from S3 bucket to Budibase internal database table.   
Automation is working as expected (i.e., read csv file from S3 buckets and adds/appends data to Buidbase internal table). 

Problem:
Every time job runs, it appends rows from CSV file creating duplicate records, hence would like to delete all rows in table before commencing job. 

Help required on how to delete all rows in Buildbase internal table (thru automation preferably)

**Comment by mikesealey** [comment, marked as answer]
Set up an automation that uses a query rows step on the table in question. This will return an array (of rows).

Add a "Delete Row" step as the next action, and set it to a looping step. You can then pass in the array from the query-rows step.

The Delete Row action needs a row-id, but you will now see in the bindings the "Loop Outputs" bindings. - make sure to add `._id` to supplu the row_id to the delete-step.
![image](https://github.com/user-attachments/assets/52a96739-65ee-4987-a2f3-d8a842c7dd31)

It might look a little bit like this, and will delete eery row in your table. Please be careful with automations like this for obvious reasons.
![image](https://github.com/user-attachments/assets/7b976fd5-a122-4b60-b5df-1eb84d076ae7)

**Reply by mallipatil** [reply]
Great, Thanks for your help.
